library/core/src/lib.rs

```diff
@@ -115,7 +114,7 @@
 #![feature(const_intrinsic_copy)]
 #![feature(const_intrinsic_forget)]
 #![feature(const_likely)]
-#![feature(const_maybe_uninit_as_ptr)]
+#![feature(const_maybe_uninit_as_mut_ptr)]
 #![feature(const_maybe_uninit_assume_init)]
```

Comment:
Can this be removed now?

Reply:
No, as the method takes a mutable parameter. `const_mut_refs` hasn't yet been stabilized.

Reply:
This is for `assume_init`, which is being stabilized... is it not? I am confused.

Reply:
Whoops, when I responded I thought misread where the code was. I'll check the necessity of this.

Reply:
Removing this results in the following errors:

```text
error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/mem/maybe_uninit.rs:915:19
    |
915 |             &mut *self.as_mut_ptr()
    |                   ^^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/ptr/mod.rs:374:32
    |
374 |         copy_nonoverlapping(x, tmp.as_mut_ptr(), 1);
    |                                ^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/ptr/mod.rs:485:17
    |
485 |         let t = t.as_mut_ptr() as *mut u8;
    |                 ^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/ptr/mod.rs:511:17
    |
511 |         let t = t.as_mut_ptr() as *mut u8;
    |                 ^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/ptr/mod.rs:701:34
    |
701 |         copy_nonoverlapping(src, tmp.as_mut_ptr(), 1);
    |                                  ^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: `maybe_uninit::MaybeUninit::<T>::as_mut_ptr` is not yet stable as a const fn
   --> library/core/src/ptr/mod.rs:793:47
    |
793 |         copy_nonoverlapping(src as *const u8, tmp.as_mut_ptr() as *mut u8, mem::size_of::<T>());
    |                                               ^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(const_maybe_uninit_as_mut_ptr)]` to the crate attributes to enable

error: could not compile `core` due to 6 previous errors
```

As such I'll be leaving the attribute in.

Reply:
I was asking about the `const_maybe_uninit_assume_init`, not the `const_maybe_uninit_as_mut_ptr`.

Reply:
Yeah, that probably should have been explicitly stated 🙂 I'll check when I'm on my laptop and send a follow-up if necessary.

Reply:
Thanks. :)

Reply:
Removing that line results in an error on line 496 of the same file (inside `MaybeUninit::write`). That's for the `assume_init_mut` method.

So...I _think_ this means that const stability isn't currently subject to the same checks, in that a feature can be both simultaneously stable and unstable? If that's the case we should fix that.

Reply:
Ah I see, makes sense. I opened an issue for this: https://github.com/rust-lang/rust/issues/91357